mantidproject/mantid
# FlatPaalmanPingsCorrection update

`FlatPlatePaalmanPingsCorrection` should now properly support `Direct` and `Indirect` modes.

**To test:**

Test the algorithm with some different parameters, particularly the `Direct`, `Indirect`, `Elastic` and `Efixed` options. See the new system tests, doc tests and unit tests for how to do this.

It would be good to check the old and new behaviours of `Efixed` and `Elastic`, which should still give comparable results. Below is an example script, using the newly added system test files. `Efixed` and `Elastic` should match to `1e-3`.

```
ws  = LoadNexus(Filename='IN16B_felo_002K.nxs')
ws = ConvertUnits(InputWorkspace=ws, Target="Wavelength", EMode="Indirect", EFixed=2.08)
val = ws.extractX()
val[:, 0:2] = 2.602
val[:, -2:] = 31.974
for i in range(18):
    ws.setX(i, val[i, :])
    
ws_can = CloneWorkspace(ws)

mode = 'Efixed'

FPPP_Result = FlatPlatePaalmanPingsCorrection(SampleWorkspace=ws, Emode=mode, Efixed=2.08,
                                              SampleChemicalFormula='V',
                                              SampleDensity=0.0704565, SampleDensityType='Number Density',
                                              SampleThickness=0.2, SampleAngle=-45,
                                              CanWorkspace=ws_can, CanChemicalFormula='Ti',
                                              CanDensity=0.0567, CanDensityType='Number Density',
                                              CanFrontThickness=0.05, CanBackThickness=0.05)

LoadNexusProcessed(Filename="FlatPlatePaalmanPings_" + mode + ".nxs", OutputWorkspace='ref')
result = CompareWorkspaces(Workspace1=FPPP_Result, Workspace2='ref', Tolerance=1e-3)
if not result[0]:
    print ("Mismatch in " + mode + ": " + result[1].row(0)['Message'])
else:
    print("Workspaces match")
```

Fixes #19959.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
